### PR TITLE
[android] Updated Elevation implementation so that it works with differents borders sizes

### PR DESCRIPTION
**Pull request by machard**
When using elevation while using the triangle css tricks, the shadow is not good.

before:
capture d ecran 2016-03-31 a 12 15 57

after:
capture d ecran 2016-04-05 a 11 21 43

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java [hunk 1]
```diff
@@ -304,6 +304,59 @@ private void updatePath() {
         : null;
   }
 
+  private Path getPathForBorderOutline() {
+    Path mPathForBorderOutline = new Path();
```
Comment (trave7er) [inline ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java]: shall we allocate this path once, and reuse it by calling reset()?

**Comment by trave7er** [comment]
@machard , thank you for catching this bug I missed. Do you think we can put some example code for this in UIExplorer? Maybe add a shadow to the example triangle? @mkonicek @dmmiller

**Comment by mkonicek** [comment]
cc @corbt

**Comment by mkonicek** [comment]
cc @kmagiera

File: ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java (hunk 1 above)
Comment (machard) [inline ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java]: ok I'll update it

**Comment by corbt** [comment]
What happens if you have a triangle that also sets a border radius on one or more borders?

I imagine that this PR doesn't make that situation worse, but is there a way we can extend the logic so it covers that as well?

**Comment by machard** [comment]
@corbt If i'm not mistakien, border radius on android will make/need all borders to have the same width.

**Comment by corbt** [comment]
@machard ok. Agree that adding a triangle with a shadow somewhere to the UIExplorer sounds like a good idea. Otherwise LGTM!

File: ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java (hunk 1 above)
Comment (mkonicek) [inline ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java]: Cool, thanks!

File: ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java [hunk 2]
```diff
@@ -143,7 +143,7 @@ public void getOutline(Outline outline) {
 
       outline.setConvexPath(mPathForBorderRadiusOutline);
     } else {
-      outline.setRect(getBounds());
+      outline.setConvexPath(getPathForBorderOutline());
```
Comment (mkonicek) [inline ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java]: Does this have impact on performance in the common case (rectangle)?

**Comment by mkonicek** [comment]
Thanks for doing this! Please add the example in the UIExplrorer, maybe in the `View` example.

**Comment by facebook-github-bot-2** [comment]
@machard do you have any updates for this pull request? It's been a while so we wanted to check in and see if you've looked at the requested changes.

**Comment by mkonicek** [comment]
@machard I'm going through all pull requests and noticed that this one hasn't been updated in a while and has some comments asking questions. I'll unfortunately close this for now (it's a useful fix!) to keep the number of open pull requests down; if you plan to continue working on it please send a new pull request linking to this one.

**Pull request closed without merging**